packages/core/src/payment/strategies/amazon-pay-v2/amazon-pay-v2-payment-processor.ts

```diff
@@ -124,6 +143,10 @@ export default class AmazonPayV2PaymentProcessor {
                       },
                   };
 
+            if (decoupleCheckoutInitiation) {
+                delete buttonOptions.createCheckoutSessionConfig;
```

Comment:
Does it make sense to delete the key? Or maybe clone it into another object with the correct interface?

Reply:
I think it makes sense since `createCheckoutSessionConfig` is optional and needs to be set depending on the flow type. And also I think is the easiest way, but of course, I can clone into another object and exclude the property.

Reply:
@animesh1987 let me know what you prefer.

Reply:
I feel cloning is better, since typescript in the latest version throws an error if delete is used.